### ISSUE TITLE
How do I call a RemoteObject method from Apache Royale?

### ISSUE THREAD

**Issue by lmarevaloa** [apache/royale-asjs#434]
I need is to add more than one method to the RemoteObject.

Using the example as3, but adding one more method, how would this look in Apache Royale?
<mx:RemoteObject id="Server" destination="Server" source="gb.informaticasystems.Server" fault="handler_backendCommunicationFails(event)" >
  <mx:method name="executeQuery" result="handler_fetchDataRequestSuccess(event)"/>
  <mx:method name="getData" result="handler_getDataSuccess(event)"/>
</mx:RemoteObject>

Someone has a similar example but in Apache Royale??

**Comment by piotrzarzycki21** [comment]
Did you check this one ? [RemoteObjectAMFTest](https://github.com/apache/royale-asjs/tree/develop/examples/royale/RemoteObjectAMFTest)

**Comment by lmarevaloa** [comment]
Yes, but what I do not understand is where I put the path of the source parameter; that routes the method file.
example source parameter:
>>>as3.
<mx:RemoteObject id="Server" destination="Server" source="gb.informaticasystems.Server" fault="handler_backendCommunicationFails(event)" >
<mx:method name="getData" result="handler_getDataSuccess(event)"/>
</mx:RemoteObject>
>>>apache royale.
 <fx:Script>
		<![CDATA[
 protected function getSomeCompressedProduct():void
			{
                Server.send("getData", []);
			}
		]]>
	</fx:Script>
<js:CompressedRemoteObject id="Server" destination="Server" result="onResult(event)" fault="onFault(event)"          endPoint = "http://localhost:8080/messagebroker/websocket-amf"
                        />

**Comment by carlosrovira** [comment]
Hi,

first thing I see is that right example to look is [this](https://github.com/apache/royale-asjs/tree/develop/examples/mxroyale/RemoteObjectAMFTest). The other one is for Royale RemoteObject that I not recommend to use. Better use MXRoyale emulation of Flex RO.

I'm using MX RO extensively, but not using `mx:method` mxml tag, although I see in that example that Alex put several ones and I remember that was working.

About "source" property. I don't recommend to use that since is deprecated and expose the internals of your server, in opposite I use `destination`, and `channelset`, that uses `endpoint`.

CompressedRemoteObject is used when you compress data in your server, so maybe is not right to use for your case.

**Comment by lmarevaloa** [comment]
Hi Carlos,
If I tried to use MXRoyale library, but it doesn't show anything I get the blank page. Do you have a solution in Jewel UI?

**Comment by carlosrovira** [comment]
Hi, I use Jewel UI with RemoteObject AMF from MXRoyale. I only use MXRoyale for that and some few more classes like some validator that is still not created in Jewel. Notice that in the project I linked before there's and example mxml App with Jewel and other without Jewel. Compile one or the other to see the diferences.

**Comment by aharui** [comment]
The goal of the MX Emulation Components is to emulate existing Flex behavior whenever possible.  The "source" property is not well documented in the Language Reference so it isn't clear how it is supposed to work.  It looks like you should be able to set it from MXML.  Whether it is deprecated or not, if it used to work, we should consider trying to make it work in Royale.

Volunteers are needed to help make it work correctly if it isn't.  Simple test cases and a test server would be helpful, not to mention an actual pull request with any required changes.

**Comment by carlosrovira** [comment]
Maybe I'm wrong, but I think others already made `source` work so don't think is broken. I recommend starting from the example I posted with your backend or using the one we have in our `examples/amf` folder to test ([this](https://github.com/apache/royale-asjs/tree/develop/examples/amf/SampleAmfWebApp)), and from a working app that sends AMF, try to tweak to use `source` property. Then report back with your findings.

**Comment by alinakazi** [comment]
We are using it this way:
responderComp1 = new Responder(fn_roCompListRH, fn_HandleFault);
				tokenComp1 = roCompList.fn_GetCompListDP(); 
				tokenComp1.addResponder(responderComp1);
                trace(tokenComp1);

instead of 
roCompList.fn_GetCompListDP();

&

	<mx:RemoteObject id="roCompList" endpoint="http://localhost:8080/SHMA/messagebroker/websocket-amf"
						 destination="shmCompanySetupBO"/>

result and fault handlers are removed from mx:RemoteObject Section.
